conda/plugins/types.py

```diff
@@ -16,13 +16,35 @@
 
 if TYPE_CHECKING:
     from argparse import ArgumentParser, Namespace
-    from typing import Callable
+    from typing import Callable, Iterable, Protocol
 
     from ..common.configuration import Parameter
     from ..core.solve import Solver
+    from ..env.env import Environment
     from ..models.match_spec import MatchSpec
     from ..models.records import PackageRecord
 
+    class CondaEnvInstallerInstallProtocol(Protocol):
```

Comment:
```suggestion
    class EnvInstallerInstallProtocol(Protocol):
```

Reply:
Do we use the word Protocol instead of Abstract now